src/macaron/slsa_analyzer/analyzer.py

```diff
@@ -536,31 +615,23 @@ def add_component(
         """
         # Note: the component created in this function will be added to the database.
         repository = None
-        if analysis_target.repo_path:
-            git_obj = self._prepare_repo(
-                os.path.join(self.output_path, self.GIT_REPOS_DIR),
-                analysis_target.repo_path,
-                analysis_target.branch,
-                analysis_target.digest,
-                analysis_target.parsed_purl,
-            )
-            if git_obj:
-                # TODO: use both the repo URL and the commit hash to check.
-                if (
-                    existing_records
-                    and (existing_record := existing_records.get(git_url.get_remote_origin_of_local_repo(git_obj)))
-                    is not None
-                ):
-                    raise DuplicateCmpError(
-                        f"{analysis_target.repo_path} is already analyzed.", context=existing_record.context
-                    )
+        if git_obj:
```

Comment:
If the type annotation of `git_obj` is not nullable, I wonder why we need to check for null-ability of `git_obj` here.